src/NHibernate/Linq/LinqExtensionMethods.cs

```diff
@@ -98,5 +98,45 @@ public static IFutureValue<T> ToFutureValue<T>(this IQueryable<T> query)
 
 			return (IFutureValue<TResult>) provider.ExecuteFuture(expression);
 		}
+
+		
+		/// <summary>
+		/// Deletes all entities in the specified query. The delete operation is performed in the database.
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <param name="query">The query.</param>
+		/// <returns></returns>
+		public static int Delete<T>(this IQueryable<T> query)
```

Comment:
Possibly. I think that Delete().Execute() would look a bit strange, with a double verb.